tesseract/tesseract_kinematics/include/tesseract_kinematics/ikfast/ikfast_inv_kin.h

```diff
@@ -0,0 +1,148 @@
+
+#include <tesseract_kinematics/core/inverse_kinematics.h>
+
+namespace tesseract_kinematics
+{
+/**
+ * @brief IKFast Inverse Kinematics Implmentation.
+ *
+ * This along with the ikfast_inv_kin.hpp is to be used with a generated ikfast to create a tesseract implementation.
+ * Once you have created your ikfast solver of your robot all you need is to create header similiar to what is shown
+ * below:
+ *
+ * Header File: fanuc_p50ib_15_inv_kinematics.h
+ *
+ * #include <Eigen/Geometry>
+ * #include <vector>
+ * #include <tesseract_kinematics/ikfast/ikfast_inv_kin.h>
+ * #include <tesseract_kinematics/core/forward_kinematics.h>
+
+ * namespace fanuc_p50ib_15_ikfast_wrapper
+ * {
+ * class FanucP50iBInvKinematics : public tesseract_kinematics::IKFastInvKin
+ * {
+ * public:
+ *   FanucP50iBInvKinematics(const std::string name,
+ *                           const std::string base_link_name,
+ *                           const std::string tip_link_name,
+ *                           const std::vector<std::string> joint_names,
+ *                           const std::vector<std::string> link_names,
+ *                           const std::vector<std::string> active_link_names,
+ *                           const Eigen::MatrixX2d joint_limits)
+ * };
+ *
+ * Cpp File:
+ *
+ * #include <tesseract_kinematics/ikfast/impl/ikfast_inv_kin.hpp>
+ * #include <fanuc_p50ib_15_ikfast_wrapper/impl/fanuc_p50ib_15_ikfast.hpp>
+ * #include <fanuc_p50ib_15_ikfast_wrapper/tesseract_fanuc_p50ib_kinematics.h>
+ *
+ * namespace fanuc_p50ib_15_ikfast_wrapper
+ * {
+ *   FanucP50iBInvKinematics::FanucP50iBInvKinematics(const std::string name,
+ *                                                    const std::string base_link_name,
+ *                                                    const std::string tip_link_name,
+ *                                                    const std::vector<std::string> joint_names,
+ *                                                    const std::vector<std::string> link_names,
+ *                                                    const std::vector<std::string> active_link_names,
+ *                                                    const Eigen::MatrixX2d joint_limits)
+ *   : FanucP50iBInvKinematics(name, base_link_name, tip_link_name, joint_names, link_names, active_link_names,
+ joint_limits)
+ *   {}
+ * }
+ *
+*/
+class IKFastInvKin : public InverseKinematics
+{
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  using Ptr = std::shared_ptr<IKFastInvKin>;
+  using ConstPtr = std::shared_ptr<const IKFastInvKin>;
+
+  IKFastInvKin(const std::string name,
+               const std::string base_link_name,
+               const std::string tip_link_name,
+               const std::vector<std::string> joint_names,
+               const std::vector<std::string> link_names,
+               const std::vector<std::string> active_link_names,
+               const Eigen::MatrixX2d joint_limits);
+
+  bool calcInvKin(Eigen::VectorXd& solutions,
```

Comment:
I realize these are pretty self explanatory, but missing Doxygen.

Reply:
Documentation is in the base class.